<a href="https://colab.research.google.com/github/nsk31512/terver/blob/hometask3/%D0%A2%D0%B5%D1%80%D0%B2%D0%B5%D1%80_%D0%94%D0%B73.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#Задача 1

Даны значения зарплат из выборки выпускников: 
```
100, 80, 75, 77, 89, 33, 45, 25, 65, 17, 30, 230, 24, 57, 55, 70, 75, 65, 84, 90, 150
```
Используя только встроенные питоновские функции и структуры данных (т.е. без библиотек `numpy`, `pandas` и др.), посчитайте (несмещённое) среднее квадратичное отклонение для данной выборки.

In [15]:
import math

In [9]:
arr = [100, 80, 75, 77, 89, 33, 45, 25, 65, 17, 30, 230, 24, 57, 55, 70, 75, 65, 84, 90, 150]
print(arr)

[100, 80, 75, 77, 89, 33, 45, 25, 65, 17, 30, 230, 24, 57, 55, 70, 75, 65, 84, 90, 150]


In [12]:
sum = 0
for el in arr:
  sum += el

avg = sum / len(arr)
avg

73.14285714285714

In [17]:
sum = 0
for el in arr:
  foo = (el - avg) ** 2  
  sum += foo

std = math.sqrt(sum / (len(arr) - 1))
std

47.34795213553139

Среднее квадратичное отклонение данной выборки 47.348

#Задача 2

Найдите число выбросов в выборке из задачи 1. Для определения выбросов используйте методику как при построении "усов" в `boxplot`, однако, как и в задаче 1, пользоваться можно только встроенными функциями и структурами данных.


In [18]:
len(arr) % 4

1

На  4  не делится. Значит, квартиль придётся на конкретный элемент, а именно элемент с индексом length // 4 (опять же поскольку индексация происходит с нуля).

In [23]:
arr_sorted = sorted(arr)
print(arr_sorted)

[17, 24, 25, 30, 33, 45, 55, 57, 65, 65, 70, 75, 75, 77, 80, 84, 89, 90, 100, 150, 230]


In [28]:
#первый квартиль
q1 = arr_sorted[len(arr) // 4]
q1

45

In [30]:
#третий квартиль
q3 = arr_sorted[3 * len(arr) // 4]
q3

84

In [32]:
iqr = q3 - q1
boxplot_range = [q1 - 1.5 * iqr, q3 + 1.5 * iqr]
boxplot_range

[-13.5, 142.5]

In [33]:
count = 0
for el in arr_sorted:
  if el < boxplot_range[0] or el > boxplot_range[1]:
    count += 1
count

2

Ответ: 2 выброса

#Задача 3

В университет на факультет A поступило 100 человек, на факультет B - 90 человек и на факультет C - 45 человек. Вероятность того, что студент с факультета A сдаст первую сессию, равна 0.6. Для студента с факультета B эта вероятность равна 0.7, а для студента с факультета C - 0.8. Случайным образом выбирается студент, сдавший первую сессию. Какое событие наиболее вероятно:
1. студент учится на факультете A,
2. студент учится на факультете B,
3. студент учится на факультете C?

_Замечание_: да, эта задача не на тему статистики, но тема важная, и её стоит иногда освежать в памяти.

Посчитаем математическое ожидание для каждого случая

In [68]:
100 * 0.6

60.0

In [69]:
90 * 0.7

62.99999999999999

In [70]:
45 * 0.8

36.0

Наибольшее значение мат. ожидания для факультета В, значит событие выбрать студанта с факультета В наиболее вероятно